### ISSUE TITLE
[data/values] support reading values from environment variables or passed-in flags

### ISSUE THREAD

**Issue by nimakaviani** [k14s/ytt#2]
the `ytt` user may wish to provide data values in the form of environment variables or by passing values through the command line. this is particularly beneficial when providing secrets.

As a user of `ytt` I expect it to provide me with either of the options above to provide or override values.

**Comment by cppforlife** [comment]
per our previous discussions:

  - multiple values are read from env vars
    - `ytt tpl ... --data-values-env BAR_`
    - BAR_FOO_KEY1_KEY2 -> Foo.Key1.Key2
  - single value is read from command arg
    - `ytt tpl ... --data-value-arg Foo.Key1.Key2=myval` (where myval is YAML unmarshable value)
  - single value is read from file
    - `ytt tpl ... --data-value-file Foo.Key1.Key2=path` (where path is some file whose contents will be loaded as a string)

**Comment by cppforlife** [comment]
as a side note, i was thinking one can "override" default values by specifying it in a dedicated overlay file, which brings an interesting question of how strict should `--data-value-arg Foo.Key1.Key2=myval` be? for example what if `Foo.Key1.Key2` does not exist in the original values file?

**Comment by nimakaviani** [comment]
I feel like the behavior should be `merge` here if the key does not exist but the values are supplied to the command line explicitly

**Comment by cppforlife** [comment]
> if the key does not exist but the values are supplied to the command line explicitly

only concern i have here is catching typos. that's why overlay functionality is conservative until it's marked as ok to be missing.

**Comment by emayssat** [comment]
I want this feature ;-) 
I would like to read values passed on the command line or in the environment.
Any ETA for this feature?

**Comment by cppforlife** [comment]
@emayssat thanks for bumping this up. ill think more about what this feature should look like this week.

**Comment by cppforlife** [comment]
@emayssat @afcastano ive added https://github.com/k14s/ytt/commit/b0760475f2eb77166744baf876ff16890a0e53eb as a first cut for this feature. try it out. examples:

```
export YTT_key__env=123
ytt template -R -f . --data-value key.subkey="value" --data-values-env YTT --data-value-file key.file=/path
ytt template -R -f . --data-value key.boolean=true
```

above keys should be specified in configuration data values, otherwise overlaying will fail, eg

```
#@data/values
---
key:
  subkey:
  boolean:
  env:
  file:
```

**Comment by cppforlife** [comment]
@emayssat @afcastano let me know if this works out for yall. have a good weekend.

**Comment by cppforlife** [comment]
included in v0.8.0
